shared/common-adapters/video.desktop.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+import * as React from 'react'
+import Measure from 'react-measure'
+import type {Props, State} from './video'
+import * as Styles from '../styles'
+import {getVideoSize, CheckURL} from './video.shared'
+
+export default class extends React.PureComponent<Props, State> {
+  state = {
+    containerHeight: 0,
+    containerWidth: 0,
+    loadedVideoSize: false,
+    videoHeight: 0,
+    videoWidth: 0,
+  }
+
+  _mounted = false
+
+  _onContainerResize = ({bounds}) =>
+    this._mounted && this.setState({containerHeight: bounds.height, containerWidth: bounds.width})
+
+  _videoRef: {current: HTMLVideoElement | null} = React.createRef()
+  _onVideoClick = () =>
+    this._videoRef.current &&
+    (this._videoRef.current.paused ? this._videoRef.current.play() : this._videoRef.current.pause())
+
+  _onVideoLoadedmetadata = ({target}) => {
+    this._mounted &&
```

Comment:
@chrisnojima I couldn't find out if syntheticEvent can be called when unmounted, so kept this here.